File: android/src/main/java/com/reactnativeawsivsplayerview/AwsIvsPlayerViewManager.java
package com.reactnativeawsivsplayerview;

import androidx.annotation.Nullable;

import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.common.MapBuilder;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;

import java.util.Map;

import javax.annotation.Nonnull;

public class AwsIvsPlayerViewManager extends SimpleViewManager<AwsIvsPlayerView> {
    public static final String REACT_CLASS = "AwsIvsPlayerView";

    @Nonnull
    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Nonnull
    @Override
    protected AwsIvsPlayerView createViewInstance(@Nonnull ThemedReactContext reactContext) {
        return new AwsIvsPlayerView(reactContext);
    }

    @Override
    public void onDropViewInstance(AwsIvsPlayerView view) {
        super.onDropViewInstance(view);
        view.cleanupMediaPlayerResources();
        view.release();
    }

//    @Override
//    @Nullable
//    public Map getExportedCustomDirectEventTypeConstants() {
//        MapBuilder.Builder builder = MapBuilder.builder();
//        for (RNRtmpView.Events event : RNRtmpView.Events.values()) {
//            builder.put(event.toString(), MapBuilder.of("registrationName", event.toString()));
//        }
//        return builder.build();
//    }

    @Override
    public Map<String,Integer> getCommandsMap() {
        MapBuilder.Builder builder = MapBuilder.builder();
        for (AwsIvsPlayerView.Commands command : AwsIvsPlayerView.Commands.values()) {
            builder.put(command.toString(), command.ordinal());
        }
        return builder.build();
    }

    @Override
    public void receiveCommand(AwsIvsPlayerView videoView, int commandId, @Nullable ReadableArray args) {
        AwsIvsPlayerView.Commands command = AwsIvsPlayerView.Commands.values()[commandId];

        switch (command){
            case COMMAND_LOAD:
                videoView.load(args.getString(0));
                break;
            case COMMAND_PAUSE:
                videoView.pause();
                break;
            default:
                break;
        }
    }
}

File: android/src/main/java/com/perrystreet/AwsIvsPlayerViewManager.kt
package com.perrystreet

import com.facebook.react.bridge.ReadableArray
import com.facebook.react.common.MapBuilder
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp
import javax.annotation.Nonnull

class AwsIvsPlayerViewManager : SimpleViewManager<AwsIvsPlayerView>() {
    @Nonnull
    override fun getName(): String {
        return REACT_CLASS
    }

    @Nonnull
    override fun createViewInstance(@Nonnull reactContext: ThemedReactContext): AwsIvsPlayerView {
        return AwsIvsPlayerView(reactContext)
    }

    override fun onDropViewInstance(view: AwsIvsPlayerView) {
        super.onDropViewInstance(view)
        view.cleanupMediaPlayerResources()
        view.release()
    }

    override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any?>? {
        val builder: MapBuilder.Builder<String, Any?> = MapBuilder.builder<String, Any?>()
        for (event in AwsIvsPlayerView.Events.values()) {
            builder.put(event.toString(), MapBuilder.of("registrationName", event.toString()))
        }
        return builder.build()
    }

    override fun getCommandsMap(): Map<String, Int>? {
        val builder: MapBuilder.Builder<String, Int> = MapBuilder.builder<String, Int>()
        for (command in AwsIvsPlayerView.Commands.values()) {
            builder.put(command.toString(), command.ordinal)
        }
        return builder.build()
    }

    override fun receiveCommand(videoView: AwsIvsPlayerView, commandId: Int, args: ReadableArray?) {
        val command = AwsIvsPlayerView.Commands.values()[commandId]
        when (command) {
            AwsIvsPlayerView.Commands.COMMAND_LOAD -> videoView.load(args!!.getString(0))
            AwsIvsPlayerView.Commands.COMMAND_PAUSE -> videoView.pause()
            AwsIvsPlayerView.Commands.COMMAND_MUTE -> videoView.mute()
            AwsIvsPlayerView.Commands.COMMAND_UNMUTE -> videoView.unMute()
            AwsIvsPlayerView.Commands.COMMAND_STOP -> videoView.stop()
            else -> {
            }
        }
    }

    @ReactProp(name = PROP_MAX_BUFFER_TIME_SECONDS)
    fun setMaxBufferTimeSeconds(videoView: AwsIvsPlayerView, bufferTimeInSeconds: Float) {
        videoView.setMaxBufferTimeInSeconds(bufferTimeInSeconds.toLong())
    }

    companion object {
        const val REACT_CLASS = "AwsIvsPlayerView"
        const val PROP_MAX_BUFFER_TIME_SECONDS = "maxBufferTimeSeconds"
    }
}